### PR TITLE
PrimTorch + nvFuser backend

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds a new AOT Autograd-based backend with the "--nvprims-nvfuser" cli argument for benchmarks.

### CODE REVIEW

File: benchmarks/common.py [hunk 1]
```diff
@@ -1401,6 +1401,12 @@ def parse_args():
     parser.add_argument(
         "--prims-nvfuser", action="store_true", help="user prims + nvfuser backend"
     )
+    parser.add_argument(
```
Comment (anijain2305) [inline benchmarks/common.py:1404]: Nit - We can directly use `--backend=nvprims_nvfuser`, if you didn't know about this option.

Comment (IvanYashchuk) [reply benchmarks/common.py:1404]: I didn't know! I will try and remove this code then.

File: torchdynamo/optimizations/training.py [hunk 2]
```diff
@@ -431,6 +471,12 @@ def create_aot_backends():
     # directly lowers to NVFuser without relying no Torchscript.
     BACKENDS["prims_nvfuser"] = aot_prims_nvfuser
```
Comment (jansel) [inline torchdynamo/optimizations/training.py:472]: Is this backend still needed?  What is the difference with `nvprims_nvfuser`?

Comment (IvanYashchuk) [reply torchdynamo/optimizations/training.py:472]: It's not needed for our team and there are no plans to use it or recommend anyone to use it. It wasn't me who added this, so I'm not going to touch it.

"prims_nvfuser" was added in https://github.com/pytorch/torchdynamo/pull/584 and it makes a decision of translating aten->prims based on a static outdated and not maintained table from https://github.com/pytorch/pytorch/blob/09965957cd8ecc696852e73022892b3ad4475783/torch/fx/passes/backends/nvfuser.py#L70-L71
The performance of this backend is terrible, on the order of 5% of eager mode's performance.

"nvprims_nvfuser" decomposes aten ops into _nvprims_, a subset of prims that are guaranteed to be executable by nvFuser, and the decomposition is used only if given aten op is fully decomposable into nvprims.

Comment (jansel) [reply torchdynamo/optimizations/training.py:472]: Let's just remove it if you don't recommend anyone use it.

cc @SherlockNoMad

**Comment by IvanYashchuk** [comment]
@anijain2305 I reverted changes to `benchmarks/common.py` to prefer `--backend=nvprims_nvfuser` argument. Other than removing "prims_nvfuser" that I'm still hesitant to do (https://github.com/pytorch/torchdynamo/pull/1281#discussion_r977379888) this PR is ready for another round of review.

**Comment by IvanYashchuk** [comment]
@anijain2305 @jansel ^

**Review by jansel — APPROVED** [review]

File: torchdynamo/optimizations/training.py (hunk 2 above)
Comment (SherlockNoMad) [reply torchdynamo/optimizations/training.py:472]: I created this backend for your team's use. 
Now that you have your own implementation, please feel free to remove this.

**Comment by IvanYashchuk** [comment]
Can someone hit the merge button?

**Comment by desertfire** [comment]
You can run `make format` to fix the lint error.

**Comment by IvanYashchuk** [comment]
@desertfire, thanks, I ran it. Could you please approve to run the workflows?

**Review by anijain2305 — APPROVED** [review]

**Comment by anijain2305** [comment]
Thanks @IvanYashchuk for the contribution. Your PR is merged.

**Pull request merged**